### PR TITLE
FIX: use lipschitz constants in prox

### PR DESCRIPTION
**Pull request by mathurinm**
Based on:
image

right @josephsalmon @qb3 ?

I'll let you compare benchmark results on main and this branch

### CODE REVIEW

**Comment by QB3** [comment]
Good for me

File: solvers/cd.py [hunk 1]
```diff
@@ -54,18 +55,21 @@ def run(self, n_iter):
                 self.X.data, self.X.indices, self.X.indptr, self.y, self.lmbd,
                 self.gamma, n_iter)
         else:
-            self.w = self.cd(self.X, self.y, self.lmbd, self.gamma, n_iter)
+            lipschitz = np.sum(self.X ** 2, axis=0)
```
Comment (agramfort) [inline solvers/cd.py:58]:
```suggestion
            lipschitz = np.linalg.norm(self.X, axis=0) ** 2
```

**Review by agramfort — APPROVED** [review]
just a suggestion

**Comment by josephsalmon** [comment]
I agree with the formula.
Good catch @mathurinm !
implicitly we were stuck with a normalized design.
This should solve the convergence issue raised in #6 .

**Pull request merged**